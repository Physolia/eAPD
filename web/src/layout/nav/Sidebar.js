import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import stickybits from 'stickybits';
import Nav from './Nav';

const Sidebar = ({ place }) => {
  useEffect(() => {
    stickybits('.site-sidebar', { stickyBitStickyOffset: 60 });
  }, []);

  const hasImage = [].indexOf(place.id) < 0;
  const imgExt = ['png', 'svg'][
    ['as', 'gu', 'mp', 'vi'].indexOf(place.id) < 0 ? 1 : 0
  ];

  return (
    <aside className="site-sidebar">
      <div className="ds-u-display--flex ds-u-align-items--center ds-u-border-bottom--1 ds-u-padding-y--2 ds-u-margin-bottom--4">
        {hasImage && (
          <img
            src={`/static/img/states/${place.id}.${imgExt}`}
            alt={place.name}
            className="ds-u-margin-right--2"
            width="40"
            height="40"
          />
        )}
        <h1>{place.name}</h1>
      </div>
      <Nav />
    </aside>
  );
};

Sidebar.propTypes = {
  place: PropTypes.object.isRequired
};

export default Sidebar;
