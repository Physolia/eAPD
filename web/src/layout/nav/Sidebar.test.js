import { shallow } from 'enzyme';
import React from 'react';

import Sidebar from './Sidebar';

describe('Sidebar component', () => {
  const props = {
    place: { id: 'place id', name: 'place name' }
  };

  it('renders correctly', () => {
    const component = shallow(<Sidebar {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('uses the PNG file extension for territories (these are not SVGs)', () => {
    const component = shallow(
      <Sidebar {...props} place={{ id: 'vi', name: 'U.S. Virgin Islands' }} />
    );
    const img = component.find('img');
    expect(img.prop('src').endsWith('.png')).toBeTruthy();
  });
});
