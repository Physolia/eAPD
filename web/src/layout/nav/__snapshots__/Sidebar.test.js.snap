// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<aside
  className="site-sidebar"
>
  <div
    className="ds-u-display--flex ds-u-align-items--center ds-u-border-bottom--1 ds-u-padding-y--2 ds-u-margin-bottom--4"
  >
    <img
      alt="place name"
      className="ds-u-margin-right--2"
      height="40"
      src="/static/img/states/place id.svg"
      width="40"
    />
    <h1>
      place name
    </h1>
  </div>
  <Connect(Nav) />
</aside>
`;
